streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala

```diff
@@ -295,9 +297,42 @@ class KStream[K, V](val inner: KStreamJ[K, V]) {
    * @return a [[KStream]] that contains more or less records with new key and value (possibly of different type)
    * @see `org.apache.kafka.streams.kstream.KStream#transform`
    */
-  def transform[K1, V1](transformerSupplier: () => Transformer[K, V, KeyValue[K1, V1]],
+  def transform[K1, V1](transformerSupplier: TransformerSupplier[K, V, KeyValue[K1, V1]],
```

Comment:
Question: should we reform the process method to also expect a `ProcessorSupplier`? It shouldn't break anyone on 2.12+ due to SAM conversion.

Reply:
I don't think we should change `process` because it would require a KIP. We should rather fix the bug only. Maybe be can create a Jira for it though.